### ISSUE TITLE
convert_labels.py warnings

### ISSUE THREAD

**Issue by MyVanitar** [martinkersner/train-DeepLab#29]
Hello,

When I want to convert my RGB labels to 1D labels, I get such warnings for each PNG file:

```
/usr/local/lib/python2.7/dist-packages/skimage/io/_io.py:132: UserWarning: train/0257.png is a low contrast image
  warn('%s is a low contrast image' % fname)
```

and when I check the PNG files, they are black.

may I ask you what this conversion does actually and the reason of this warning and black PNG labels?

**Comment by martinkersner** [comment]
Hi @VanitarNordic,

Class labels are from 0 to 20 (same as values in label images), therefore using basic image viewer cannot show you differences between classes, all seems to be black and this is also reason for warning.

Martin

**Comment by MyVanitar** [comment]
Actually I have modified the `utils.py` for different number of classes and colors. My dataset has 2 classes but anyway I don't think it would matter.

My another question is: what this conversion does on an image? Does it make the image indexed?

**Comment by martinkersner** [comment]
@VanitarNordic 
Image labels in Pascal VOC are distributed as RGB (3-channel) images, however, training algorithm requires as input 1-channel image labels. So, for example class 15 has assigned RGB color as (192, 128, 128), but for training we need to replace pixels of such color with number 15.

Yes, it makes image indexed.

**Comment by VanitarNordic** [comment]
Hello Martin,

Thank you for the clarification.

Please have a look at this image (second images). it seems the model misunderstood the ignore_label around objects as a class. Why this happened? 
Does the `ignore_label` have the 255 value after re-indexing the labels?

![screenshot from 2017-04-28 17-45-54](https://cloud.githubusercontent.com/assets/18719591/25530898/2396308c-2c3d-11e7-820f-a9d5e6f392bf.png)

**Comment by VanitarNordic** [comment]
@martinkersner 

Thank you.

Do you know any tool to help me to generate PNG segmentation labels in the VOC format?

**Comment by VanitarNordic** [comment]
@martinkersner 

Would you please tell me what modification should I make to convert PNGs with different number of classes? in my case I have 2 classes.

I modified `pascal_classes` and `pascal_palette` , but the generated files seems to have just one value instead of 3 (background, void, object)

**Comment by MyVanitar** [comment]
@martinkersner 

I have some difficulties to detect small object with DeepLabV2. I use this repo: https://github.com/DrSleep/tensorflow-deeplab-resnet

I reached to 93% IOU on my own dataset, but the model in unable to detect some small objects inside the validation set and some big objects do not have very fine edges, so the results are under question for me and too good to be true. 
I'm curious about the DeepLabV2 itself or might be the repo is buggy. **one big help from you would be that you send me one of your PNG labels to check it with my labels.**  I really appreciate it sir.
